lib/paypro.js

```diff
@@ -124,25 +124,29 @@ PayPro.get = function(opts, cb) {
     var decodedDetails = BitcorePayPro.PaymentDetails.decode(serializedDetails);
     var pd = new BitcorePayPro(COIN);
     pd = pd.makePaymentDetails(decodedDetails);
+    
+    var network = pd.get('network') == 'test' ? 'testnet' : 'livenet';
 
     var outputs = pd.get('outputs');
-    if (outputs.length > 1)
-      return cb(new Error('Payment Protocol Error: Requests with more that one output are not supported'))
+    var outs = outputs.map(function(output){
 
-    var output = outputs[0];
+      let amount = output.get('amount').toNumber();
 
-    var amount = output.get('amount').toNumber();
-    var network = pd.get('network') == 'test' ? 'testnet' : 'livenet';
+      // We love payment protocol
+      let offset = output.get('script').offset;
+      let limit = output.get('script').limit;
 
-    // We love payment protocol
-    var offset = output.get('script').offset;
-    var limit = output.get('script').limit;
+      // NOTE: For some reason output.script.buffer
+      // is only an ArrayBuffer
+      let buffer = new Buffer(new Uint8Array(output.get('script').buffer));
+      let scriptBuf = buffer.slice(offset, limit);
+      let addr = new bitcore.Address.fromScript(new bitcore.Script(scriptBuf), network);
+      return {addr: addr.toString(), amount: amount}
+    });
 
-    // NOTE: For some reason output.script.buffer
-    // is only an ArrayBuffer
-    var buffer = new Buffer(new Uint8Array(output.get('script').buffer));
-    var scriptBuf = buffer.slice(offset, limit);
-    var addr = new bitcore.Address.fromScript(new bitcore.Script(scriptBuf), network);
+    var totalAmount = outs.reduce(function (a, b) {
```

Comment:
Name the anonymous function.

Reply:
are you referring to the callback function being passed as an argument to reduce?

Reply:
That's correct. The existing code doesn't always have that, but that's what we are doing going forward. It helps by expressing your intention and also provides better information in the call stack when debugging.

Reply:
Changes made